ocean_lib/models/v4/test/test_factory_router.py

```diff
@@ -0,0 +1,444 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+from re import A
+import hexbytes
+from toolz.itertoolz import cons
+from web3.main import Web3
+from tests.resources.helper_functions import (
+    get_publisher_wallet,
+    get_consumer_wallet,
+    get_another_consumer_wallet,
+)
+
+import pytest
+
+from web3 import exceptions
+from ocean_lib.web3_internal.wallet import Wallet
+
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.erc721_token import ERC721Token
+from ocean_lib.models.v4.factory_router import FactoryRouter
+from ocean_lib.web3_internal.contract_utils import get_contracts_addresses
+from ocean_lib.models.v4.bfactory import BFactory
+from ocean_lib.models.v4.erc20_token import ERC20Token
+from ocean_lib.models.v4.models_structures import ErcCreateData
+from ocean_lib.web3_internal.wallet import Wallet
+from tests.resources.helper_functions import get_factory_deployer_wallet
+from ocean_lib.config import Config
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+
+
+import os
+
+_NETWORK = "ganache"
+_NETWORK = "ganache"
+
+
+def get_factory_router_address(config):
+    """Helper function to retrieve a known factory router address."""
+    return get_contracts_addresses(address_file=config.address_file, network=_NETWORK)[
+        "Router"
+    ]
+
+
+def init(web3, config):
+    factory_router = FactoryRouter(web3, get_factory_router_address(config))
+    factory_router.update_opf_fee(
+        web3.toWei(0.001, "ether"), get_factory_deployer_wallet(network=_NETWORK)
+    )
+
+
+def create_new_token(web3, config):
+    tkn = ERC20Token.deploy(web3=web3, deployer_wallet=get_publisher_wallet())
+    return tkn
+
+
+def get_erc_address(config):
+    """Helper function to retrieve a known ERC address."""
+    return get_contracts_addresses(address_file=config.address_file, network=_NETWORK)[
+        "ERC20Template"
+    ]["1"]
+
+
+def get_ocean_address(config):
```

Comment:
I will add an utils file from where you can retrieve those functions that get the address for a specific contract instead of creating them in all the test files.